### ISSUE TITLE
feature: [SECURITY]: don't read secrets from env vars

### ISSUE THREAD

**Issue by happysalada** [windmill-labs/windmill#2473]
environment variables are available to other processes, storing secrets in them is discouraged.
This is why docker has a special mechanism for secrets called docker secrets. (systemd has credentials).
I understand security is not a priority when trying to start a company, so feel free to leave this open until this can be tackled.
The change should not be too involved though. from what I see, the main secret that is exposed through an env var is DATABASE_URL.
If there was a corresponding environment variable that was named DATABASE_URL_FILE, that would just store a file path that would read the content of the path and that would be the database url. (docker secrets and systemd credentials both work with a file path).
The tricky part I think is the migration from using a file and from reading directly from the environment. How would you prefer this to be handled ? Just add another environment variable and require either or to be set ? (that would mean that there is still a way to have an insecure setup, but this leaves more time for users to migrate).
For context, I'm trying to add a systemd service for nixos. (the binary is already packaged, now we need a service to run this simply).
happy to elaborate on anything if you need.

Thanks again for starting such a great repo!

**Comment by rubenfiszel** [comment]
Thank for the issue,

Just to be clear, we are clearing the env variables before starting any job so DATABASE_URL is not available to child jobs. Additionally, we provide nsjail for complete sandboxing of jobs on EE. Security is a very important topic for us and we are currently deployed in environment with very tight security requirements.

 Agreed that it is still suboptimal security wise  to pass secrets in env variables, since some other processes can read them from /proc/ and it is not as convenient to properly permission that vs doing a pointer to a file as you suggest and secure the file itself. 

Indeed, in the absence of DATABASE_URL we could check for existence of DATABASE_URL_FILE and do as you suggest. It's a small PR. If you volunteer to do it, it would be awesome, otherwise we will add it to our backlog.

Thanks a lot ofr the hard work of packaging it for nixos. I'm a big fan but never had time yet to explore it properly.
